src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -591,6 +590,142 @@ def get_default_realm(self) -> str:
+        if len(zonegroup_list) > 0:
+            for zonegroup in zonegroup_list:
+                cmd_zonegroup_option = ['--rgw-zonegroup', zonegroup]
+                rgw_zonegroup_info_cmd = ['zonegroup', 'get']
+                rgw_zonegroup_info_cmd += cmd_zonegroup_option
+                try:
+                    _, out, err = mgr.send_rgwadmin_command(rgw_zonegroup_info_cmd)
+                    if out:
+                        all_zonegroup_info.append(out)
+                        if out['id'] == default_zonegroup_id:
+                            defaults_info['default_zonegroup'] = out['name']
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+        rgw_zone_get_cmd = ['zone', 'get'] 
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_zone_get_cmd)
+            if out:
+                defaults_info['default_zone'] = out['name']      
+        except SubprocessError as error:
+                raise DashboardException(error, http_status_code=500, component='rgw')
+        multisite_info.append(all_realms_info)
+        multisite_info.append(all_zonegroup_info)
+        multisite_info.append(defaults_info)
+        return multisite_info
+    
+    
+    def create_realm(self, params: dict):
+        realm_name = params.get('realm_name')
+        default = params.get('default')
+        rgw_realm_create_cmd = ['realm', 'create']
+        cmd_create_realm_options = ['--rgw-realm', realm_name]
+        if default:
+            cmd_create_realm_options.append('--default')
+        rgw_realm_create_cmd += cmd_create_realm_options
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_realm_create_cmd)
+        except SubprocessError as error:
+            raise DashboardException(error, http_status_code=500, component='rgw')
+        return out
+            
+    def create_zonegroup(self, params: dict):
+        realm_name = params.get('realm_name')
+        zonegroup_name = params.get('zonegroup_name')
+        default = params.get('default')
+        master = params.get('master')
+        endpoints = params.get('endpoints')
+        rgw_zonegroup_create_cmd = ['zonegroup', 'create']
+        cmd_create_zonegroup_options = ['--rgw-zonegroup', zonegroup_name]
+        if realm_name is not None:
+            cmd_create_zonegroup_options.append('--rgw-realm')
+            cmd_create_zonegroup_options.append(realm_name)
+        if default:
+            cmd_create_zonegroup_options.append('--default')
+        if master:
+            cmd_create_zonegroup_options.append('--master')
+        if endpoints is not None:
+            cmd_create_zonegroup_options.append('--endpoints')
+            cmd_create_zonegroup_options.append(endpoints)
+        rgw_zonegroup_create_cmd += cmd_create_zonegroup_options
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_zonegroup_create_cmd)
```

Comment:
It might be worth checking errors that a command might return, I'm not sure all errors are raised.

Reply:
Yes, am doing this currently, will update the PR with that